### ISSUE TITLE
[Task] frontend/ implement accessibility features

### ISSUE THREAD

**Issue by ArdaYalcindag** [bounswe/bounswe2025group5#562]
### Task Description
The [SRS](https://github.com/bounswe/bounswe2025group5/wiki/Requirements-of-CMPE451) is updated according to [WCAG 2.1 AA](https://github.com/bounswe/bounswe2025group5/wiki/WC3-Standards#1-web-content-accessibility-guidelines-wcag-21-aa)  guidelines and related requirements are needed to be implemented. The related issues are 2.1.2.5, 2.1.2.6, 2.1.2.7, 2.1.2.8, 2.1.2.9, 2.4.3, 2.4.4. Their sub-issues should be opened and implemented.

### Why Is This Needed?
To achieve a more accessible web application to better include users from all backgrounds.

### Acceptance Criteria
- [ ] 2.1.2.5
- [ ] 2.1.2.6
- [ ] 2.1.2.7
- [ ] 2.1.2.8
- [ ] 2.1.2.9
- [ ] 2.4.3
- [x] 2.4.4

# Additional Notes
feature-specific sub issues should be opened and added under this issue.

### Deadline
23.11.2025 - 23.59
